src/mcts/node.h

```diff
@@ -428,13 +413,24 @@ class NodeTree {
   Node* GetGameBeginNode() const { return gamebegin_node_.get(); }
   const PositionHistory& GetPositionHistory() const { return history_; }
 
+  // Not thread safe!
+  void UpdateDepth(uint16_t new_node_depth);
+  uint16_t GetMaxDepth() const;
+  float GetAverageDepth() const;
+
  private:
   void DeallocateTree();
+
+  void RecalculateDepth(); // Brute force re-examines the entire tree
+
   // A node which to start search from.
   Node* current_head_ = nullptr;
   // Root node of a game tree.
   std::unique_ptr<Node> gamebegin_node_;
   PositionHistory history_;
+  uint16_t max_depth_ = 0;
```

Comment:
Could you remind me why it is not in `Search`?

Reply:
Because the max depth of the `Search` need not be the max depth of the `NodeTree`.